cirq-core/cirq/transformers/routing/route_circuit_cqc_test.py

```diff
@@ -107,6 +107,45 @@ def test_circuit_with_measurement_gates():
     cirq.testing.assert_same_circuits(routed_circuit, circuit)
 
 
+def test_circuit_with_valid_intermediate_multi_qubit_measurement_gates():
+    device = cirq.testing.construct_ring_device(3)
+    device_graph = device.metadata.nx_graph
+    router = cirq.RouteCQC(device_graph)
+    q = cirq.LineQubit.range(3)
+    hard_coded_mapper = cirq.HardCodedInitialMapper({q[i]: q[i] for i in range(3)})
+
+    valid_circuit = cirq.Circuit(cirq.H.on_each(*q), cirq.measure_each(*q))
+
+    c_routed = router(
+        valid_circuit, initial_mapper=hard_coded_mapper, context=cirq.TransformerContext(deep=True)
+    )
+    device.validate_circuit(c_routed)
+
+
+def test_circuit_with_invalid_intermediate_multi_qubit_measurement_gates():
+    device = cirq.testing.construct_ring_device(3)
+    device_graph = device.metadata.nx_graph
+    router = cirq.RouteCQC(device_graph)
+    q = cirq.LineQubit.range(3)
+    hard_coded_mapper = cirq.HardCodedInitialMapper({q[i]: q[i] for i in range(3)})
+
+    invalid_circuit = cirq.Circuit(
+        cirq.H.on_each(*q),
+        cirq.MeasurementGate(3).on(*q),
+        cirq.H.on_each(*q),
+        cirq.MeasurementGate(3).on(*q),
+    )
+
+    with pytest.raises(
+        ValueError, match="Non-terminal measurements on three or more qubits are not supported"
```

Comment:
this test is brittle. it will break if we decide to change the error message. instead you have one of 2 options
1. no `match`
2. match with a regular expression (e.g. `match=r"Non-terminal.*"`)

Either is fine with me.